self-core-impl/src/test/java/com/selfxdsd/core/projects/StripeWalletITCase.java

```diff
@@ -0,0 +1,146 @@
+            Mockito.when(invoice.createdAt()).thenReturn(now);
+            Mockito.when(invoice.isPaid()).thenReturn(false);
+            Mockito.when(invoice.totalAmount()).thenReturn(BigDecimal.TEN);
+
+            final Contract contract = Mockito.mock(Contract.class);
+            final Contributor contributor = Mockito.mock(Contributor.class);
+            Mockito.when(contract.contributor()).thenReturn(contributor);
+            Mockito.when(invoice.contract()).thenReturn(contract);
+
+            final Storage storage = Mockito.mock(Storage.class);
+            final PayoutMethods allPayoutsMethods = Mockito
+                .mock(PayoutMethods.class);
+            final PayoutMethods payoutsOfContrib = Mockito
+                .mock(PayoutMethods.class);
+            final PayoutMethod payoutMethod = Mockito.mock(PayoutMethod.class);
+            final PaymentMethods allPaymentMethods = Mockito
+                .mock(PaymentMethods.class);
+            final PaymentMethods paymentsOfWallet = Mockito
+                .mock(PaymentMethods.class);
+            final PaymentMethod paymentMethod = Mockito
+                .mock(PaymentMethod.class);
+
+            final Invoices invoices = Mockito.mock(Invoices.class);
+            Mockito.when(storage.invoices()).thenReturn(invoices);
+
+            final Project project = Mockito.mock(Project.class);
+            final Wallets allWallets = Mockito.mock(Wallets.class);
+            final Wallets ofProject = Mockito.mock(Wallets.class);
+
+            final Wallet stripe = new StripeWallet(
+                storage,
+                project,
+                BigDecimal.valueOf(1000),
+                "123StripeID",
+                Boolean.TRUE,
+                "stripe_24343"
+            );
+
+            Mockito.when(storage.payoutMethods()).thenReturn(allPayoutsMethods);
+            Mockito.when(allPayoutsMethods.ofContributor(contributor))
+                .thenReturn(payoutsOfContrib);
+            Mockito.when(payoutsOfContrib.active()).thenReturn(payoutMethod);
+            Mockito.when(payoutMethod.identifier()).thenReturn("ac_123");
+
+            Mockito.when(storage.paymentMethods())
+                .thenReturn(allPaymentMethods);
+            Mockito.when(allPaymentMethods.ofWallet(stripe))
+                .thenReturn(paymentsOfWallet);
+            Mockito.when(paymentsOfWallet.active()).thenReturn(paymentMethod);
+            Mockito.when(paymentMethod.identifier()).thenReturn("pm_123");
+
+            Mockito.when(storage.wallets()).thenReturn(allWallets);
+            Mockito.when(allWallets.ofProject(project)).thenReturn(ofProject);
+
+            stripe.pay(invoice);
+
+            Mockito.verify(invoices, Mockito.times(1))
+                .registerAsPaid(Mockito.any(Invoice.class));
+            Mockito.verify(ofProject, Mockito.times(1))
+                .updateCash(stripe, BigDecimal.valueOf(990));
+        }
```

Comment:
@criske I don't think there is any way to write an efficient test for interaction with Stripe... the SDK always calls the prod URL

Reply:
@amihaiemil Just needed a "successful" status to verify the Invoice payment logic.

Reply:
@criske right, my bad. I missed the "overrideApiBase" method